pkg/controllers/secret/secretsync.go

```diff
@@ -0,0 +1,159 @@
+			// TODO: report error after some while?
+			klog.V(2).Infof("Waiting for secret %s from cloud-credentials-operator", util.CloudCredentialSecretName)
+			return nil
+		}
+		return err
+	}
+
+	driverSecret, err := c.translateSecret(cloudSecret)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = resourceapply.ApplySecret(c.kubeClient.CoreV1(), c.eventRecorder, driverSecret)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) translateSecret(cloudSecret *v1.Secret) (*v1.Secret, error) {
+	content, ok := cloudSecret.Data[cloudSecretKey]
+	if !ok {
+		return nil, fmt.Errorf("OpenStack credentials secret %s did not contain key %v", util.CloudCredentialSecretName, cloudSecretKey)
+	}
+
+	var clouds clientconfig.Clouds
+	err := yaml.Unmarshal(content, &clouds)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal clouds credentials stored in secret %s: %v", util.CloudCredentialSecretName, err)
+	}
+
+	cloud, ok := clouds.Clouds[cloudName]
+	if !ok {
+		return nil, fmt.Errorf("failed to parse clouds credentials stored in secret %s: cloud %s not found", util.CloudCredentialSecretName, cloudName)
+	}
+
+	data := make(map[string][]byte)
+
+	if cloud.AuthInfo.AuthURL != "" {
+		data["os-authURL"] = []byte(cloud.AuthInfo.AuthURL)
+	}
+	if cloud.RegionName != "" {
+		data["os-region"] = []byte(cloud.RegionName)
+	}
+	if cloud.AuthInfo.UserID != "" {
+		data["os-userID"] = []byte(cloud.AuthInfo.UserID)
+	} else if cloud.AuthInfo.Username != "" {
+		data["os-userName"] = []byte(cloud.AuthInfo.Username)
+	}
+	if cloud.AuthInfo.Password != "" {
+		data["os-password"] = []byte(cloud.AuthInfo.Password)
+	}
+	if cloud.AuthInfo.ProjectID != "" {
+		data["os-projectID"] = []byte(cloud.AuthInfo.ProjectID)
+	} else if cloud.AuthInfo.ProjectName != "" {
+		data["os-projectName"] = []byte(cloud.AuthInfo.ProjectName)
+	}
+	if cloud.AuthInfo.DomainID != "" {
+		data["os-domainID"] = []byte(cloud.AuthInfo.DomainID)
+	} else if cloud.AuthInfo.DomainName != "" {
+		data["os-domainName"] = []byte(cloud.AuthInfo.DomainName)
+	}
+	if cloud.AuthInfo.ProjectDomainID != "" {
+		data["os-projectDomainID"] = []byte(cloud.AuthInfo.ProjectDomainID)
+	} else if cloud.AuthInfo.ProjectDomainName != "" {
+		data["os-projectDomainName"] = []byte(cloud.AuthInfo.ProjectDomainName)
+	}
+	if cloud.AuthInfo.UserDomainID != "" {
+		data["os-userDomainID"] = []byte(cloud.AuthInfo.UserDomainID)
+		data["os-domainID"] = []byte(cloud.AuthInfo.UserDomainID)
+	} else if cloud.AuthInfo.UserDomainName != "" {
+		data["os-userDomainName"] = []byte(cloud.AuthInfo.UserDomainName)
+		data["os-domainName"] = []byte(cloud.AuthInfo.UserDomainName)
+	}
+	if cloud.CACertFile != "" {
+		data["os-certAuthorityPath"] = []byte(cloud.CACertFile)
+	}
+
+	secret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.ManilaSecretName,
+			Namespace: util.OperatorNamespace,
+		},
+		Type: v1.SecretTypeOpaque,
+		Data: data,
+	}
+
+	return &secret, nil
```

Comment:
Side note: why does the CSI driver and the cluster-credentials-operator produce distinct secrets? Are both formats "officially" supported/valid in OpenStack? If so, I'd like to see either cloud-credentials-operator or the CSI driver supporting an additional format. But that's not a discussion for this PR...

Reply:
I talked to @Fedosin and there are at least three different confing/secret formats in OpenStack, each component using its own. I agree it should be in cloud-credentials-operator